src/main/java/org/ngafid/flights/process/CSVFileProcessor.java

```diff
@@ -46,105 +56,353 @@ private CSVFileProcessor(Connection connection, byte[] bytes, String filename, P
+        if (headerLines.get(1) != null) {
 
-            for (String name : doubleTimeSeries.keySet()) {
-                LOG.info("name = " + name + "; = " + doubleTimeSeries.get(name));
+            String[] secondLineArray = headerLines.get(1).split(",", -1);  // Use -1 to preserve empty values
+            String[] thirdLineArray = headerLines.get(2).split(",", -1);    // Use -1 to preserve empty values
+
+            if (secondLineArray.length != thirdLineArray.length) {
+                throw new IllegalArgumentException("Lines have different lengths.");
+            }
+
+            for (int i = 0; i < secondLineArray.length; i++) {
+                String processedDataType = CSVFileProcessor.extractContentInsideParentheses(secondLineArray[i].strip());
+                dataTypes.add(processedDataType);  // Add the processed result to dataTypes
+                String processedHeader = thirdLineArray[i].strip();
+                headers.add(processedHeader.isEmpty() ? "<Empty>" : processedHeader);  // Add header or "<Empty>" if it's blank
             }
+        }
+    }
 
-        } catch (IOException | FatalFlightFileException | CsvException e) {
-            e.printStackTrace();
+    /**
+     * Calculates md5Hash from a given flight (flight entries)
+     * @param flight
+     * @return md5Hash
+     * @throws FlightProcessingException
+     */
+    private String calculateMd5Hash(List<String[]> flight) throws FlightProcessingException {
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            StringBuilder flightDataBuilder = new StringBuilder();
+            // Append each cell from the flight data to the builder
+            for (String[] flightRow : flight) {
+                for (String cell : flightRow) {
+                    flightDataBuilder.append(cell);
+                }
+            }
+            // Calculate MD5 hash and convert to hex
+            byte[] flightHash = md.digest(flightDataBuilder.toString().getBytes(StandardCharsets.UTF_8));
+            String result = DatatypeConverter.printHexBinary(flightHash).toLowerCase();
+            System.out.println("Md5 hash calculated: " + result);
+            return result;
+        } catch (NoSuchAlgorithmException e) {
             throw new FlightProcessingException(e);
         }
+    }
+
+    /**
+     * Extracts three header lines from a csv file, removes # from the beginning of the second line (if # is present)
+     * @param bufferedReader
+     * @return List of header lines (3 lines expected)
+     * @throws IOException
+     */
+    private List<String> extractHeaderLines(BufferedReader bufferedReader) throws IOException {
+        String flightInformationLine = bufferedReader.readLine();
+        String secondLine = bufferedReader.readLine();
+        String thirdLine = bufferedReader.readLine();
+
+        if(secondLine.startsWith("#")){
```

Comment:
need spaces e.g. `if (...) {`

Reply:
This applies to a lot of code but im not going to mark them all.

Reply:
Done, thanks!